src/main/java/com/megastore/repository/ProductRepository.java

```diff
@@ -14,4 +15,78 @@ public interface ProductRepository extends BaseRepository<Product> {
+            "AND p.product_brand = :brand " +
+            "ORDER BY p.updated DESC " +
+            "LIMIT :limit ",
+            nativeQuery = true)
+    Collection<Product> findAll(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    @Query(value = "SELECT * " +
+            "FROM product p " +
+            "JOIN subcategories s ON p.subcategory_id = s.id " +
+            "JOIN categories c ON s.category_id = c.id " +
+            "JOIN images img ON img.product_id = p.id " +
+            "WHERE s.id = :subcategoryId " +
+            "AND p.product_price BETWEEN :priceFrom AND :priceTo " +
+            "AND p.product_brand = :brand " +
+            "ORDER BY p.product_title DESC, p.updated DESC " +
+            "LIMIT :limit ",
+            nativeQuery = true)
+    Collection<Product> findAllSortedByNameDESC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    @Query(value = "SELECT * " +
+            "FROM product p " +
+            "JOIN subcategories s ON p.subcategory_id = s.id " +
+            "JOIN categories c ON s.category_id = c.id " +
+            "JOIN images img ON img.product_id = p.id " +
+            "WHERE s.id = :subcategoryId " +
+            "AND p.product_price BETWEEN :priceFrom AND :priceTo " +
+            "AND p.product_brand = :brand " +
+            "ORDER BY p.product_title ASC, p.updated DESC " +
+            "LIMIT :limit ",
+            nativeQuery = true)
+    Collection<Product> findAllSortedByNameASC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    @Query(value = "SELECT * " +
+            "FROM product p " +
+            "JOIN subcategories s ON p.subcategory_id = s.id " +
+            "JOIN categories c ON s.category_id = c.id " +
+            "JOIN images img ON img.product_id = p.id " +
+            "WHERE s.id = :subcategoryId " +
+            "AND p.product_price BETWEEN :priceFrom AND :priceTo " +
+            "AND p.product_brand = :brand " +
+            "ORDER BY p.product_price DESC, p.updated DESC " +
+            "LIMIT :limit ",
+            nativeQuery = true)
+    Collection<Product> findAllSortedByPriceDESC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
+    @Query(value = "SELECT * " +
+            "FROM product p " +
+            "JOIN subcategories s ON p.subcategory_id = s.id " +
+            "JOIN categories c ON s.category_id = c.id " +
+            "JOIN images img ON img.product_id = p.id " +
+            "WHERE s.id = :subcategoryId " +
+            "AND p.product_price BETWEEN :priceFrom AND :priceTo " +
+            "AND p.product_brand = :brand " +
+            "ORDER BY p.product_price ASC, p.updated DESC " +
+            "LIMIT :limit ",
+            nativeQuery = true)
+    Collection<Product> findAllSortedByPriceASC(int limit, long subcategoryId, Double priceFrom, Double priceTo, String brand);
+
```

Comment:
You should feel like something went wrong when you write almost identical code 6 times.  Please, find better solution.